pilot/pkg/features/pilot.go

```diff
@@ -618,7 +618,7 @@ var (
 
 	StripHostPort = env.Register("ISTIO_GATEWAY_STRIP_HOST_PORT", false,
 		"If enabled, Gateway will remove any port from host/authority header "+
-			"before any processing of request by HTTP filters or routing. Deprecated: in Istio 1.15+ port is ignored in domain matching.").Get()
```

Comment:
This means actually port was skipped

Reply:
According to this [pr](https://github.com/envoyproxy/envoy/pull/14491), it will mutate the original host header.

I also deploy a envoy to approve this. The following is the envoy static config.
```
- name: envoy.filters.network.http_connection_manager
        typed_config:
          "@type": type.googleapis.com/envoy.extensions.filters.network.http_connection_manager.v3.HttpConnectionManager
          scheme_header_transformation:
            scheme_to_overwrite: https
          strip_any_host_port: true
          stat_prefix: ingress_http
          route_config:
            name: local_route
            virtual_hosts:
            - name: local_service
              domains: ["*"]
              routes:
              - match:
                  prefix: "/"
                route:
                  cluster: outbound|80||httpbin.static
          http_filters:
          - name: envoy.filters.http.rbac
            typed_config:
              "@type": type.googleapis.com/envoy.extensions.filters.http.rbac.v3.RBAC
              rules:
                action: ALLOW
                policies:
                  "test":
                     permissions:
                     - any: true
                     principals:
                     - header:
                         name: ":authority"
                         string_match:
                           exact: "test.com"
          - name: envoy.filters.http.router
```

Using curl to call `curl -v -H "host: test.com:5333" localhost:10000/ip`. Then we can see envoy log.
```
envoy_1    | [2023-03-25 08:23:14.536][20][debug][rbac] [source/extensions/filters/http/rbac/rbac_filter.cc:68] checking request: requestedServerName: , 
sourceIP: 172.27.0.1:38336, 
directRemoteIP: 172.27.0.1:38336, 
remoteIP: 172.27.0.1:38336,
localAddress: 172.27.0.3:10000, 
ssl: none, 
headers: ':authority', 'test.com'
envoy_1    | ':path', '/ip'
envoy_1    | ':method', 'GET'
envoy_1    | ':scheme', 'https'
envoy_1    | 'user-agent', 'curl/7.68.0'
envoy_1    | 'accept', '*/*'
envoy_1    | 'x-forwarded-proto', 'https'
envoy_1    | 'x-request-id', 'fe8c18e5-3258-46ce-ad77-eef338452930'
```

The original host `test.com:5333` has been updated to test.com.